hw/xbox/xbox.c

```diff
@@ -227,11 +227,62 @@ static void xbox_memory_init(PCMachineState *pcms,
     xbox_flash_init(rom_memory);
 }
 
+uint8_t *load_eeprom(void)
+{
+    char *filename;
+    int fd;
+    int rc;
+    int eeprom_file_size;
+    const int eeprom_size = 256;
+
+    uint8_t *eeprom_data = g_malloc(eeprom_size);
+
+    const char *eeprom_file = object_property_get_str(qdev_get_machine(),
+                                                      "eeprom", NULL);
+    if ((eeprom_file != NULL) && *eeprom_file) {
+        filename = qemu_find_file(QEMU_FILE_TYPE_BIOS, eeprom_file);
+        assert(filename);
+
+        eeprom_file_size = get_image_size(filename);
+        if (eeprom_size != eeprom_file_size) {
+            fprintf(stderr,
+                    "qemu: EEPROM file size != %d bytes. (Is %d bytes)\n",
+                    eeprom_size, eeprom_file_size);
+            g_free(filename);
+            exit(1);
+            return NULL;
+        }
+
+        fd = open(filename, O_RDONLY | O_BINARY);
+        if (fd < 0) {
+            fprintf(stderr, "qemu: EEPROM file '%s' could not be opened.\n", filename);
+            g_free(filename);
```

Comment:
Not `free`ing `eeprom_data`, but it doesn't really matter here because we are exiting.

Reply:
Should still be considered a leak, no?

Reply:
Yes, and some cleanup should fix it eventually because it just polutes valgrind logs.